collections/windowed_test.ts

```diff
@@ -0,0 +1,220 @@
+    windowed(numbers, 3);
+    assertEquals(numbers, [1, 2, 3, 4, 5]);
+  },
+});
+
+Deno.test({
+  name: "[collections/windowed] empty input",
+  fn() {
+    windowedTest([[], 3], []);
+    windowedTest([[], 3, {}], []);
+    windowedTest([[], 3, { step: 2 }], []);
+    windowedTest([[], 3, { partial: true }], []);
+    windowedTest([[], 3, { step: 2, partial: true }], []);
+  },
+});
+
+Deno.test({
+  name: "[collections/windowed] default option",
+  fn() {
+    windowedTest([[1, 2, 3, 4, 5], 5], [
+      [1, 2, 3, 4, 5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 3], [
+      [1, 2, 3],
+      [2, 3, 4],
+      [3, 4, 5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 1], [
+      [1],
+      [2],
+      [3],
+      [4],
+      [5],
+    ]);
+  },
+});
+
+Deno.test({
+  name: "[collections/windowed] step was set",
+  fn() {
+    windowedTest([[1, 2, 3, 4, 5], 5, { step: 2 }], [
+      [1, 2, 3, 4, 5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 3, { step: 2 }], [
+      [1, 2, 3],
+      [3, 4, 5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 1, { step: 2 }], [
+      [1],
+      [3],
+      [5],
+    ]);
+  },
+});
+
+Deno.test({
+  name: "[collections/windowed] partial was set",
+  fn() {
+    windowedTest([[1, 2, 3, 4, 5], 5, { partial: true }], [
+      [1, 2, 3, 4, 5],
+      [2, 3, 4, 5],
+      [3, 4, 5],
+      [4, 5],
+      [5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 3, { partial: true }], [
+      [1, 2, 3],
+      [2, 3, 4],
+      [3, 4, 5],
+      [4, 5],
+      [5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 1, { partial: true }], [
+      [1],
+      [2],
+      [3],
+      [4],
+      [5],
+    ]);
+  },
+});
+
+Deno.test({
+  name: "[collections/windowed] step and partial was set",
+  fn() {
+    windowedTest([[1, 2, 3, 4, 5], 5, { step: 2, partial: true }], [
+      [1, 2, 3, 4, 5],
+      [3, 4, 5],
+      [5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 3, { step: 2, partial: true }], [
+      [1, 2, 3],
+      [3, 4, 5],
+      [5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 1, { step: 2, partial: true }], [
+      [1],
+      [3],
+      [5],
+    ]);
+  },
+});
+
+Deno.test({
+  name: "[collections/windowed] size or step was invalid",
+  fn() {
+    assertThrows(
+      () => {
+        windowed([1, 2, 3, 4, 5], 0);
+      },
+      RangeError,
+      "Both size and step must be positive.",
+    );
+    assertThrows(
+      () => {
+        windowed([1, 2, 3, 4, 5], 3, { step: 0 });
+      },
+      RangeError,
+      "Both size and step must be positive.",
+    );
+    assertThrows(
+      () => {
+        windowed([1, 2, 3, 4, 5], NaN);
+      },
+      RangeError,
+      "Both size and step must be positive.",
+    );
+    assertThrows(
+      () => {
+        windowed([1, 2, 3, 4, 5], 3, { step: NaN });
+      },
+      RangeError,
+      "Both size and step must be positive.",
+    );
+    assertThrows(
+      () => {
+        // @ts-ignore: for test
+        windowed([1, 2, 3, 4, 5], "invalid");
```

Comment:
I don't think we need to test for invalid TS calls - if we start doing that, you might as well check every parameter.

Reply:
If we forgot to throw an error here, the `length` will be `NaN` and we will get the unexpected behavior of returning an empty array. I'd like to leave this test alone to make sure that an error is thrown when passing a non-numeric value.